group/assignment5/assignment5_graph.py

```diff
@@ -0,0 +1,38 @@
+from collections import defaultdict
+from os.path import commonprefix
+ 
+class Graph:
+    def __init__(self, dictionary):
+        self.graph = defaultdict(list)
+        self.letters = set()
+        for first, second in zip(dictionary, dictionary[1:]):
+            prefix = commonprefix([first, second])
+            first_letter = first[len(prefix)]
+            second_letter = second[len(prefix)]
+            self.draw_edge(first_letter, second_letter)
+            self.letters = self.letters | set(first) | set(second)
+ 
+    def draw_edge(self, x, y):
+        self.graph[x].append(y)
+
+    def dfs(self, node, visited, alphabet):
```

Comment:
The same as above, I would recommend to not mix up alphabet-related logic with graph implementation. In this case you can just rename the variable alphabet.